RC: RDI in the cloud (#570)
redis/docs: content/operate/rc/databases/rdi/_index.md

---
Title: Data Integration
alwaysopen: false
categories:
- docs
- operate
- rc
description: Use Redis Data Integration with Redis Cloud.
hideListLinks: true
weight: 99
---

Redis Cloud now supports [Redis Data Integration (RDI)]({{<relref "integrate/redis-data-integration">}}), a fast and simple way to bring your data into Redis.

> The applications that write data to your primary database are not always the same application that need faster access to the data. Redis Cloud uses [ingest]({{<relref "/integrate/redis-data-integration/ingest/">}}) to help you offload all read queries from the application database to Redis.

Reviewer: What you've said is correct but RDI works even if it's basically always the same app doing reads and writes.
```suggestion
A relational database usually handles queries much more slowly than a Redis database. If your application uses a relational database and makes many more reads than writes (which is the typical case) then you can improve performance by using Redis as a cache to handle the read queries quickly. Redis Cloud uses [ingest]({{<relref "/integrate/redis-data-integration/ingest/">}}) to help you offload all read queries from the application database to Redis automatically.
```

RDI's helps Redis customers sync Redis Cloud with live data from their primary databases to:

>     - Meet the required speed and scale of read queries and provide an excellent and predictable user experience.
>     - Save resources and time when building pipelines and coding data transformations.
>     - Reduce the total cost of ownership by saving money on expensive database read replicas.

Reviewer: This hasn't formatted correctly. I think you have to do it without the indent for the first nesting level of the list.
```suggestion
- Meet the required speed and scale of read queries and provide an excellent and predictable user experience.
- Save resources and time when building pipelines and coding data transformations.
- Reduce the total cost of ownership by saving money on expensive database read replicas.
```

Using RDI with Redis Cloud simplifies managing your data integration pipeline. No need to worry about hardware or underlying infrastructure, as Redis Cloud manages that for you. Creating the data flow from source to target is much easier, and there are validations in place to reduce errors.

## Get started

> To start creating your new data pipeline, see [Prepare source database]({{<relref "content/operate/rc/databases/rdi/setup.md">}}).

Reviewer: Doesn't work with the .md suffix. BTW, if you're using VSCode then maybe I can interest you in this [handy extension](https://drive.google.com/file/d/1h1h_hVFNYDOSRspkr3Epgv1Ap5Pi_2YR/view?usp=sharing) to generate relrefs? ;-)
```suggestion
To start creating your new data pipeline, see [Prepare source database]({{<relref "content/operate/rc/databases/rdi/setup">}}).
```

The applications that write data to your primary database are not always the same application that need faster access to the data. Redis Cloud uses [ingest]({{<relref "/integrate/redis-data-integration/ingest/">}}) to help you offload all read queries from the application database to Redis.

RDI's helps Redis customers sync Redis Cloud with live data from their primary databases to:
    - Meet the required speed and scale of read queries and provide an excellent and predictable user experience.
    - Save resources and time when building pipelines and coding data transformations.
    - Reduce the total cost of ownership by saving money on expensive database read replicas.

Using RDI with Redis Cloud simplifies managing your data integration pipeline. No need to worry about hardware or underlying infrastructure, as Redis Cloud manages that for you. Creating the data flow from source to target is much easier, and there are validations in place to reduce errors.

## Get started

To start creating your new data pipeline, see [Prepare source database]({{<relref "content/operate/rc/databases/rdi/setup.md">}}).

## Limitations

Only the following are supported at this time:

> - One source database connected to one target database

Reviewer 2: I would go with these limitations in this order:

- Redis Cloud Pro tier
- Ingest use case
- AWS platform (source and target database)
- One source database connected to one target database

---
Title: Data Integration
alwaysopen: false
categories:
- docs
- operate
- rc
description: Use Redis Data Integration with Redis Cloud.
hideListLinks: true
weight: 99
---

Redis Cloud now supports [Redis Data Integration (RDI)]({{<relref "integrate/redis-data-integration">}}), a fast and simple way to bring your data into Redis from other types of primary databases.

A relational database usually handles queries much more slowly than a Redis database. If your application uses a relational database and makes many more reads than writes (which is the typical case) then you can improve performance by using Redis as a cache to handle the read queries quickly. Redis Cloud uses [ingest]({{<relref "/integrate/redis-data-integration/">}}) to help you offload all read queries from the application database to Redis automatically.

> Having a data pipeline lets you have a cache that is always ready for queries. RDI Data pipelines ensure that any changes made to your primary database are captured in your Redis cache within a few seconds, preventing cache misses and stale data within the cache. 

Reviewer: Minor, but "having lets you have" sounds a bit odd. Maybe "Using a data pipeline..." or "With a data pipeline you have.." or something like that?

Reviewer 3: ```suggestion
Using a data pipeline lets you have a cache that is always ready for queries. RDI Data pipelines ensure that any changes made to your primary database are captured in your Redis cache within a few seconds, preventing cache misses and stale data within the cache. 
```
```suggestion
Having a data pipeline lets you have a cache that is always ready for queries. RDI Data pipelines ensure that any changes made to your primary database are captured in your Redis cache within a few seconds, preventing cache misses and stale data within the cache. 
```

- Reduce the total cost of ownership by saving money on expensive database read replicas.

Using RDI with Redis Cloud simplifies managing your data integration pipeline. No need to worry about hardware or underlying infrastructure, as Redis Cloud manages that for you. Creating the data flow from source to target is much easier, and there are validations in place to reduce errors.

## Data pipeline architecture

A RDI data pipeline sits between your source database and your target Redis database. Initially, the pipeline reads all of the data and imports it into the target database during the *initial cache loading* phase. After this initial sync is complete, the data pipeline enters the *change streaming* phase, where changes are captured as they happen. Changes in the source database are added to the target within a few seconds of capture.

RDI encrypts all network connections with TLS or mTLS and stores all state and configuration data inside the Redis cluster. 

For more info on how RDI works, see [RDI Architecture]({{<relref "/integrate/redis-data-integration/architecture">}}).

## Prerequisites

Before you can create a data pipeline, you must have:

> - A [Redis Cloud Pro database]() hosted on Amazon Web Services (AWS)

Reviewer 3: Yup, I'll fix these before this goes out.

Using a data pipeline lets you have a cache that is always ready for queries. RDI Data pipelines ensure that any changes made to your primary database are captured in your Redis cache within a few seconds, preventing cache misses and stale data within the cache. 

RDI helps Redis customers sync Redis Cloud with live data from their primary databases to:
- Meet the required speed and scale of read queries and provide an excellent and predictable user experience.
- Save resources and time when building pipelines and coding data transformations.
- Reduce the total cost of ownership by saving money on expensive database read replicas.

Using RDI with Redis Cloud simplifies managing your data integration pipeline. No need to worry about hardware or underlying infrastructure, as Redis Cloud manages that for you. Creating the data flow from source to target is much easier, and there are validations in place to reduce errors.

> ## Data pipeline architecture

Reviewer 3: I'm pretty sure I cover the "backfill" process by using the term "ingest" and the "tracking changes" by talking about "change streaming" - are these meaningfully different terms?

> ## Data pipeline architecture

Reviewer 3: Added Hash/JSON in [129d6af](https://github.com/redis/docs/pull/570/commits/129d6afed8282bc9f2008e378742f69781751d5b).

Connections to the source database are through [AWS PrivateLink](https://aws.amazon.com/privatelink/), ensuring that the data pipeline is only exposed to the specific database endpoint. See [Set up connectivity]({{<relref "/operate/rc/databases/rdi/setup#set-up-connectivity">}}) to learn how to connect your PrivateLink to the Redis Cloud VPC.

RDI encrypts all network connections with TLS. The pipeline will process data from the source database in-memory and write it to the target database using a TLS connection. There are no external connections to your data pipeline except from Redis Cloud management services.

## Prerequisites

Before you can create a data pipeline, you must have:

- A [Redis Cloud Pro database]({{< relref "/operate/rc/databases/create-database/create-pro-database-new" >}}) hosted on Amazon Web Services (AWS)
- One supported source database, also hosted on AWS and connected to [AWS PrivateLink](https://aws.amazon.com/privatelink/):

>     - MySQL

Reviewer 3: Fixed in [49d1f54](https://github.com/redis/docs/pull/570/commits/49d1f5491cf3794483ec822059fc2994f1594c04)

    - Oracle
    - SQL Server
    - PostgreSQL

{{< note >}}
Please be aware of the following limitations:

- The target database must be a Redis Cloud Pro database hosted on Amazon Web Services (AWS). Redis Cloud Essentials databases and databases hosted on Google Cloud do not support Data Integration.
- Only the Ingest use case is supported at this time.
- Source databases must also be hosted on AWS.
- One source database can only be connected to one target database.
{{< /note >}} 

## Get started

To create a new data pipeline, you need to:

> 1. [Prepare your source database]({{<relref "/operate/rc/databases/rdi/setup">}}) and any associated credentials.

Reviewer 3: Yes, that page also includes PrivateLink and creating the credentials secret.

Using a data pipeline lets you have a cache that is always ready for queries. RDI Data pipelines ensure that any changes made to your primary database are captured in your Redis cache within a few seconds, preventing cache misses and stale data within the cache. 

RDI helps Redis customers sync Redis Cloud with live data from their primary databases to:
- Meet the required speed and scale of read queries and provide an excellent and predictable user experience.
- Save resources and time when building pipelines and coding data transformations.
- Reduce the total cost of ownership by saving money on expensive database read replicas.

Using RDI with Redis Cloud simplifies managing your data integration pipeline. No need to worry about hardware or underlying infrastructure, as Redis Cloud manages that for you. Creating the data flow from source to target is much easier, and there are validations in place to reduce errors.

## Data pipeline architecture

> A RDI data pipeline sits between your source database and your target Redis database. Initially, the pipeline reads all of the data and imports it into the target database during the *initial cache loading* phase. After this initial sync is complete, the data pipeline enters the *change streaming* phase, where changes are captured as they happen. Changes in the source database are added to the target within a few seconds of capture. The data pipeline translates relational database rows to Redis hashes or JSON documents. 

Reviewer 4: ```suggestion
An RDI data pipeline sits between your source database and your target Redis database. Initially, the pipeline reads all of the data and imports it into the target database during the *initial cache loading* phase. After this initial sync is complete, the data pipeline enters the *change streaming* phase, where changes are captured as they happen. Changes in the source database are added to the target within a few seconds of capture. The data pipeline translates relational database rows to Redis hashes or JSON documents. 
```

> A RDI data pipeline sits between your source database and your target Redis database. Initially, the pipeline reads all of the data and imports it into the target database during the *initial cache loading* phase. After this initial sync is complete, the data pipeline enters the *change streaming* phase, where changes are captured as they happen. Changes in the source database are added to the target within a few seconds of capture. The data pipeline translates relational database rows to Redis hashes or JSON documents. 

Reviewer 4: "initial cache loading" is a wording that we never use, so it would be strange to use it here.
I suggest either:
* initial data loading
* Initial sync
* Initial synchronisation

@yaronp68 WDYT?

> A RDI data pipeline sits between your source database and your target Redis database. Initially, the pipeline reads all of the data and imports it into the target database during the *initial cache loading* phase. After this initial sync is complete, the data pipeline enters the *change streaming* phase, where changes are captured as they happen. Changes in the source database are added to the target within a few seconds of capture. The data pipeline translates relational database rows to Redis hashes or JSON documents. 

Reviewer 4: "change streaming phase" - again it's a bit weird phrasing.
I suggest either:
* change**s** streaming phase
* streaming phase
* change data capture phase (this is the official naming in the industry it seems)

@yaronp68 WDYT?

For more info on how RDI works, see [RDI Architecture]({{<relref "/integrate/redis-data-integration/architecture">}}).

### Pipeline security

> Data pipelines are set up to ensure a high level of data security. Source database credentials and TLS secrets are stored in AWS secret manager and shared using the AWS Secret Manager CSI driver for secrets. See [Share source database credentials]({{<relref "/operate/rc/databases/rdi/setup#share-source-database-credentials">}}) to learn how to share your source database credentials and TLS certificates with Redis Cloud.

Reviewer 4: As far as I know, there is no such thing like `AWS Secret Manager CSI driver`. There is a Kubernetes CSI driver and probably that's what was meant here? Secrets are coming from the AWS Secret Manager to RDI and RDI accepts these secrets through the Kubernetes CSI driver.

Reviewer 3: ```suggestion
Data pipelines are set up to ensure a high level of data security. Source database credentials and TLS secrets are stored in AWS secret manager and shared using the Kubernetes CSI driver for secrets. See [Share source database credentials]({{<relref "/operate/rc/databases/rdi/setup#share-source-database-credentials">}}) to learn how to share your source database credentials and TLS certificates with Redis Cloud.
```
